maintenance/image-options/imageset.md

```diff
@@ -0,0 +1,227 @@
+
+OPERATOR_IMAGE={{ operator.registry }}/{{ operator.image }}:{{ operator.version }}
+echo "Pulling $OPERATOR_IMAGE"
+echo
+docker pull $OPERATOR_IMAGE -q >/dev/null
+versions=$(docker run $OPERATOR_IMAGE --version)
+ver=$(echo -e "$versions" | grep 'Calico:')
+
+imagelist=($(docker run $OPERATOR_IMAGE --print-images=list))
+
+cat > ./imageset.yaml <<EOF
+apiVersion: operator.tigera.io/v1
+kind: ImageSet
+metadata:
+  name: calico-$(echo $ver | sed -e 's|^.*: *||')
+spec:
+  images:
+EOF
+
+for x in "${imagelist[@]}"; do
+  for y in ${images[*]}; do
+    if [[ $x =~ $y: ]]; then
+      echo "Pulling $x"
+      docker pull $x -q > /dev/null
+      {% raw %}digests=$(docker inspect $x -f '{{range .RepoDigests}}{{printf "%s\n" .}}{{end}}'){% endraw %}
+      name=$(echo $x | sed -e 's|docker.io/||' -e 's|:.*$||')
+      imgdigest=$(echo -e $digests | grep $name)
+      digest=$(echo -e $imgdigest | sed -e 's|^.*@||')
+      echo "Adding digest for $x"
+      echo 
+      echo "  - image: \"$(echo $x | sed -e 's|^.*/\([^/]*/[^/]*\):.*$|\1|')\"" >> ./imageset.yaml
+      echo "    digest: \"$digest\"" >> ./imageset.yaml
+    fi
+  done
+done
+```
+%>
+{% endtabs %}
+
+Apply the created `imageset.yaml` to your cluster.
+
+#### Verification steps
+
+1. Check tigerastatus for components that are Degraded with `kubectl get tigerastatus`.
+   - If any components show Degraded, [investigate further](#how-can-i-tell-if-there-is-a-problem-with-my-imageset).
+2. When tigerastatus for all components show Available True, the ImageSet has been applied.
+   ```
+   NAME     AVAILABLE   PROGRESSING   DEGRADED   SINCE
+   calico   True        False         False      54s
+   ```
+3. Verify that the correct ImageSet is being used. In Installation status, check that the `imageset` field is set to the ImageSet you created.
+   Check the field by running the following command:
+   ```
+   kubectl get installation default -o yaml | grep imageSet
+   ```
+   You should see output similar to:
+   ```
+       imageSet: calico-{{site.data.versions.first.title}}
+   ```
+
+#### Create new ImageSet when upgrading or downgrading
+
+Before upgrading to a new release or downgrading, you must create a new [ImageSet]({{site.baseurl}}/reference/installation/api#operator.tigera.io/v1.ImageSet)
+with updated image references and names for the new release. This must be done prior
+to upgrading the cluster so when the new manifests are applied, the appropriate ImageSet is available.
+
+#### Why does the Installation Resource status not include my ImageSet?
+
+The [status.imageset]({{site.baseurl}}/reference/installation/api#operator.tigera.io/v1.InstallationStatus) field of
+the Installation Resource will not be updated until the `calico` component has fully been deployed. `calico` is
+fully deployed when `kubectl get tigerastatus calico` reports Available True with Progressing and Degraded as False.
+
+#### How can I tell if there is a problem with my ImageSet?
+
```

Comment:
add commas here:

If any components are degraded, you can...
If the digest provided for an image is incorrect or unable to be pulled, the tigerastatus will not directly report that information, but...
If you suspect an issue with a resource rollout due to an issue with an image, you...